tests/integration/test_types_integration.py

```diff
@@ -0,0 +1,314 @@
+import math
+import pytest
+from decimal import Decimal
+import datetime
+from datetime import timezone, timedelta
+import trino
+
+
+@pytest.fixture
+def trino_connection(run_trino):
+    _, host, port = run_trino
+
+    yield trino.dbapi.Connection(
+        host=host, port=port, user="test", source="test", max_attempts=1
+    )
+
+
+def test_int_types(trino_connection):
+    cur = trino_connection.cursor(experimental_python_types=True)
+    cur.execute("""
+    SELECT false AS col_bool
+  , cast(null AS BOOLEAN) AS col_bool_null
+  , cast(127 AS TINYINT) AS col_tinyint
+  , cast(-128 AS TINYINT) AS col_tinyint_min
+  , cast(null AS TINYINT) AS col_tinyint_null
+  , cast(32767 AS SMALLINT) AS col_smallint
+  , cast(-32768 AS SMALLINT) AS col_smallint_min
+  , cast(null AS SMALLINT) AS col_smallint_null
+  , cast(2147483647 AS INTEGER) AS col_integer
+  , cast(-2147483648 AS INTEGER) AS col_integer_min
+  , cast(null AS INTEGER) AS col_integer_null
+  , cast(9223372036854775807 AS BIGINT) AS col_bigint
+  , cast(-9223372036854775808 AS BIGINT) AS col_bigint_min
+  , cast(null AS BIGINT) AS col_bigint_null
+      """)
+    result = cur.fetchall()
+    compare_results(result[0], [
+        False,
+        None,
+        127,
+        -128,
+        None,
+        32767,
+        -32768,
+        None,
+        2147483647,
+        -2147483648,
+        None,
+        9223372036854775807,
+        -9223372036854775808,
+        None
```

Comment:
Can you introduce a helper class/method that is similar to [SqlDataTypeTest](https://github.com/trinodb/trino/blob/master/testing/trino-testing/src/main/java/io/trino/testing/datatype/SqlDataTypeTest.java) in Trino? It will improve the readability a lot.

Reply:
Do you mean a class/method where you only pass the input SQL and expected output, or something different?

Reply:
"only pass the input SQL and expected output" is right.

The below is Trino example using `SqlDataTypeTest`. There're 4 arguments in `addRoundTrip()`, but it will be 3 arguments in this Python test.
```
    @Test
    public void testBit()
    {
        SqlDataTypeTest.create()
                .addRoundTrip("bit", "b'1'", BOOLEAN, "true")
                .addRoundTrip("bit", "b'0'", BOOLEAN, "false")
                .addRoundTrip("bit", "NULL", BOOLEAN, "CAST(NULL AS BOOLEAN)")
                .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.test_bit"));
    }
```

Reply:
How about something like:

```
SqlTest()
    .add_field("false AS col_bool", False)
    .add_field("cast(null AS BOOLEAN) AS col_bool_null", None)
    ...
    .execute()
```

Reply:
Personally, I would like to omit column names because it unrelates to the test purpose. When we want to clarify the value meaning, just adding code comment seems enough to me.

Reply:
Fair enough. But aside from that is this the type of API you had in mind?

Reply:
Yes :)

Reply:
Done